lib/EarlyFunctionAnalysis/DetectABI.cpp

```diff
@@ -754,12 +746,15 @@ void DetectABI::propagatePrototypesInFunction(model::Function &Function) {
     if (not WritesSP and not WritesCalleeArgs and not WritesCalleeReturnValues
         and WritesOnlyRegisters) {
 
-      revng_log(Log,
-                "Overwriting " << Entry.toString() << " prototype ("
-                               << Function.Prototype().toString()
-                               << ") with wrapped function's prototype: "
-                               << Prototype.toString());
-      Function.Prototype() = Prototype;
+      std::string LogMessage = "Overwriting " + Entry.toString()
+                               + " prototype ";
+      if (!Function.Prototype().empty())
+        LogMessage += "(" + serializeToString(Function.Prototype()) + ") ";
+      LogMessage += "with wrapped function's prototype: "
+                    + serializeToString(*Prototype);
+      revng_log(Log, LogMessage);
```

Comment:
1. Wrap this in `if (Log.isEnabled)`
2. Do not use `revng_log`
3. `<< DoLog` to actually "commit" the log entry

Reply:
Done.